core/src/main/java/com/linecorp/armeria/server/ServerMetrics.java

```diff
@@ -153,6 +161,14 @@ void decreaseActiveConnections() {
         activeConnections.decrementAndGet();
     }
 
+    void increasePendingResponses() {
+        pendingResponses.increment();
+    }
+
+    void decreasePendingResponses() {
+        pendingResponses.decrement();
+    }
```

Comment:
The number of `pendingResponses` seems the same as `activeRequests`. If so, we don't need to record the same value twice in different places.

Should we remove `inc()` and `dec()` in `GracefulShutdownSupport` and use `activeRequests()` for `"armeria.server.pending.responses"`

Reply:
The current changes are a clean-up but do not provide additional benefits to users. 

I prefer to add metrics for transient requests in `ServerMetrics`
- Expose a meter with `state=transient` tag for `TransientService`
- Exclude transient requests from `activeRequests`
- Switch `state=pending` to `state=active` or `state=transient`
- Add useful methods
  ```java
  public long transientHttp1Requests() {
      return transientHttp1Requests.longValue();
  }

  public long transientHttp2Requests() {
      return transientHttp2Requests.longValue();
  }
  
  public long transientRequests() {
      return transientHttp1Requests() + transientHttp2Requests();
  }

  public long allRequests() {
      return pendingRequests() + activeRequests() + transientRequests();
  }
  ```

After those changes, `pendingResponses` will be the same as `activeRequests`.

Reply:
@ikhoon That's a good suggestion. 👍